code/iaas/logic/src/main/java/io/cattle/platform/process/externalevent/ExternalEventCreate.java

```diff
@@ -161,6 +163,10 @@ public void doWithLockNoResult() {
                     spData.put(ObjectMetaDataManager.ACCOUNT_FIELD, event.getAccountId());
                     spData.put(FIELD_ZONE_ID, 1L);
 
+                    Agent agent = objectManager.findOne(Agent.class, AGENT.ACCOUNT_ID, event.getReportedAccountId(),
```

Comment:
We should adjust the agentId when the pool is found to deal with new agents taking over ownership of the pool